include/circt/Dialect/Arc/Ops.td

```diff
@@ -243,6 +243,16 @@ def MemoryWriteOp : ArcOp<"memory_write", [
   let hasVerifier = 1;
 }
 
+//===----------------------------------------------------------------------===//
+// Miscellaneous
+//===----------------------------------------------------------------------===//
+
+def TapOp : ArcOp<"tap"> {
+  let summary = "A tracker op to observe a value under a given name";
+  let arguments = (ins AnyInteger:$value, StrAttr:$name);
```

Comment:
I would restrict this to `AnySignlessInteger` since we don't use any others, right?

Reply:
Great point!